stix2validator/validator.py

```diff
@@ -274,9 +302,39 @@ def run_validation(options):
     except NoJSONFileFoundError as e:
         output.error(e.message)
 
-    results = {}
-    for fn in files:
-        results[fn] = validate_file(fn, options)
+    results = [validate_file(fn, options) for fn in files]
+
+    return results
+
+
+def validate(in_, options=None):
+    """
+    Validate objects from JSON data.  This can be an object or a list of
+    objects.
+
+    :param in_: A textual stream of JSON data.
+    :param options: Validation options
+    :return: A list of ObjectValidationResults.
+    """
+    if not options:
+        options = ValidationOptions()
+
+    obj_json = json.load(in_)
+
+    results = []
+    try:
+        if isinstance(obj_json, list):
+            for obj in obj_json:
+                results.append(validate_instance(obj, options))
+        else:
+            results.append(validate_instance(obj_json, options))
+
+    except SchemaInvalidError as ex:
+        error_result = ObjectValidationResults(is_valid=False,
+                                               fatal=ValidationErrorResults(ex))
+        results.append(error_result)
+        output.info("Data was schema-invalid. No further validation "
+                    "will be performed.")
```

Comment:
This is a new function I added.  I thought there was unnecessary code duplication, e.g. between `validate_file` and `validate_string`.  A common way to design code that processes data in such a way that it doesn't care where the data came from, is to design it to read/write from/to a stream.  I think there was a part of the design here that didn't need to care where the data came from.  That also gives flexibility to users: they can choose where the data comes from, rather than relying on us to provide a bunch of source-specific functions (e.g. validate_file, validate_string, validate_from_network, validate_from_bytebuffer, etc).  There can be one central validation function (for JSON data that isn't already parsed anyway; I think we need an analogously behaving function for parsed JSON), and source-specific validation functions can just be convenience wrappers around that.  I could cut almost all the code from `validate_string` since it's a very simple wrapper now.